relay-server/src/actors/healthcheck.rs

```diff
@@ -173,14 +208,17 @@ pub enum IsHealthy {
     Readiness,
 }
 
-/// All the message types which can be sent to the [`Healthcheck`] actor.
-#[derive(Clone, Debug)]
-pub enum HealthcheckMessage {
-    Health(IsHealthy),
-}
+impl ServiceMessage<Healthcheck> for IsHealthy {
+    type Response = bool;
 
-impl From<IsHealthy> for HealthcheckMessage {
-    fn from(is_healthy: IsHealthy) -> Self {
-        Self::Health(is_healthy)
+    fn into_envelope(self) -> (HealthcheckEnvelope, oneshot::Receiver<Self::Response>) {
+        let (tx, rx) = oneshot::channel();
+        (HealthcheckEnvelope::IsHealthy(self, tx), rx)
     }
 }
+
+/// All the message types which can be sent to the [`Healthcheck`] actor.
+#[derive(Debug)]
+pub enum HealthcheckEnvelope {
+    IsHealthy(IsHealthy, oneshot::Sender<bool>),
+}
```

Comment:
Ideally, we benchmark whether it's faster to just generally box the messages rather than having an explicit enum type. I'm still very much in favor of having the enum like this, since that takes a lot of hidden magic away and makes it very clear what's happening.

My expectation is that boxing the messages will actually improve performance. We could box them internally here, though that will increase the boilerplate.

The simpler way requiring less boiler plate would be just having a trait object, however.

Reply:
yep, this is a followup task